quadtree.js

```diff
@@ -273,74 +322,31 @@ class QuadTree {
     return found;
   }
 
-  closest(point, count, maxDistance) {
+  closest(point, count = 1, maxDistance = Infinity, furthestFound = Infinity, foundSoFar = 0) {
     if (typeof point === "undefined") {
       throw TypeError("Method 'closest' needs a point");
     }
-    if (typeof count === "undefined") {
-      count = 1;
-    }
 
-    // Limit to number of points in this QuadTree
-    if (this.length == 0) {
-      return [];
-    }
-    if (this.length < count) {
-      return this.points;
-    }
+    var inRangePoints = [...this.points].filter((p) => p.distanceFrom(point) <= maxDistance);
```

Comment:
Is there an opportunity here to calculate `furthestDistance` inside of this filter?

Reply:
Ah. Good idea. Yeah, I'll give it a go.